### ISSUE TITLE
Bug: RulerTester InvalidTestCase type - expected errors can just be number of errors

### ISSUE THREAD

**Issue by CarsonF** [typescript-eslint/typescript-eslint#6269]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```typescript
new ESLintUtils.RuleTester({...}).run('...', rule, {
  invalid: [
    { errors: 1 }
  ]
});
```

### Expected Result

No type error. ESLint has allowed this since 1.0+

https://github.com/eslint/eslint/blame/main/lib/rule-tester/rule-tester.js#L18
https://github.com/eslint/eslint/commit/7c497998c7b878014dd037ada2a372284b5c88ce#diff-5b137993d382a97ec165342bb2ece6475649983e76f0c765848adc2980b5e443R18

**Comment by bradzacher** [comment]
This is done intentionally to enforce best practices.
You should not assert with a number because it doesn't really assert anything at all.

In older versions of ESLint it also counted crashes as errors - I.e. Your rule could hard crash and the test would pass!

You should be testing with an array describing the messageIDs you expect (and data if your message is parameterised) to ensure the rule is actually working as expected.

**Comment by CarsonF** [comment]
Thanks for clarifying @bradzacher.
Merry Christmas!
